proto/vulcanize/registry/v1beta1/attributes.proto

```diff
@@ -30,4 +32,172 @@ message WebsiteRegistrationRecord {
   string tls_cert_cid       = 4 [(gogoproto.moretags) = "json:\"TLSCertCID\" yaml:\"TLSCertCID\""];
   string type               = 5 [(gogoproto.moretags) = "json:\"type\" yaml:\"type\""];
   string version            = 6 [(gogoproto.moretags) = "json:\"version\" yaml:\"version\""];
+}
+
+message GitRepository{
+  string name =1  [(gogoproto.moretags) = "json:\"name\" yaml:\"name\""];
+  string repo_reference=2 [(gogoproto.moretags) = "json:\"repo_reference\" yaml:\"repo_reference\""];
+  string description=3  [(gogoproto.moretags) = "json:\"description\" yaml:\"description\""];
+  string version            = 4 [(gogoproto.moretags) = "json:\"version\" yaml:\"version\""];
+}
+
+message Binary{
+  string raw_multicodec=1;
+  string raw_binary=2;
+  string targeted_arch=3;
+  string runtime_version=4;
+  string repo_reference=5;
+  string version=6;
+  string type=7;
+}
+
+message DockerImage{
+  string image_id=1;
+  string image_binary=2;
+  string reference_repo=3;
+  string version=4;
+  string type=5;
+}
+
+message WatcherRegistrationRecord{
+  WatcherMetadata metadata =1;
+  string repo_reference=2;
+  WASMBinary wasm=3;
+  string version=4;
+  string type=5;
+
+  message WatcherMetadata{
+    string version=1;
+    string chain_reference=2;
+  }
+
+  message WASMBinary{
+    string reference=1;
+    WASMBinaryMetadata metadata=2;
+  }
+  
+  message WASMBinaryMetadata{
+    string compiler_version=1;
+    string execution_engine_version=2;
+  }
+}
+
+message ResponderContract{
+  string service_provider_ref=1;
+  string auction_ref=2;
+  string watcher_ref=3;
+  string version=4;
+  string type=5;
+}
+
+message NPMPackage{
+  string repo_reference=1;
```

Comment:
This will reference a registered `GitRepository` record by CID/multihash

`GitRepository` also has a `repo_reference` field which will contain a qualitative/non-content-hash-reference e.g. a github or gitea web url.

It might be helpful to signal in the naming of the various `reference` fields which references are content-hash => content references vs one's that are not. E.g. `hash_reference` vs `reference`.

Reply:
There is also a npm_package_reference (CID) in this message. Should we go with renaming repo_reference with repo_hash_reference and npm_package_reference with js_package_hash_ref?

Reply:
We can avoid adding `hash` and instead we should revert to using a `/` sub-mapping to denote CID/link